components/apim-analytics/org.wso2.analytics.solutions.apim.analytics/widgets/APIMApiCreated/src/APIMApiCreated.jsx

```diff
@@ -0,0 +1,372 @@
+                    resolve();
+                })
+                .catch(error => reject(error));
+        });
+    }
+
+    /**
+     * Formats the siddhi query
+     * @memberof APIMApiCreated
+     * */
+    assembletotalQuery() {
+        if (this.state.providerConfig) {
+            const dataProviderConfigs = _.cloneDeep(this.state.providerConfig);
+            dataProviderConfigs.configs.config.queryData.query = dataProviderConfigs.configs.config.queryData.totalQuery;
+            super.getWidgetChannelManager().subscribeWidget(this.props.id, this.handleTotalCountReceived, dataProviderConfigs);
+        }
+    }
+
+    /**
+     * Formats data received from assembletotalQuery
+     * @param {object} message - data retrieved
+     * @memberof APIMApiCreated
+     * */
+    handleTotalCountReceived(message) {
+        if (message.data.length !== 0) {
+            let [[totalCount]] = message.data;
+            totalCount = totalCount < 10 ? ('0' + totalCount).slice(-2) : totalCount;
+            this.setState({ totalCount });
+        }
+        super.getWidgetChannelManager().unsubscribeWidget(this.props.id);
+        this.assembleweekQuery();
+    }
+
+    /**
+     * Formats the siddhi query using selected options
+     * @memberof APIMApiCreated
+     * */
+    assembleweekQuery() {
+        const weekStart = Moment().subtract(7, 'days');
+
+        if (this.state.providerConfig) {
+            const dataProviderConfigs = _.cloneDeep(this.state.providerConfig);
+            let query = dataProviderConfigs.configs.config.queryData.weekQuery;
+            query = query
+                .replace('{{weekStart}}', Moment(weekStart).format('YYYY-MM-DD HH:mm:ss.SSSSSSSSS'));
+            dataProviderConfigs.configs.config.queryData.query = query;
+            super.getWidgetChannelManager().subscribeWidget(this.props.id, this.handleWeekCountReceived, dataProviderConfigs);
+        }
+    }
+
+    /**
+     * Formats data received from assembleweekQuery
+     * @param {object} message - data retrieved
+     * @memberof APIMApiCreated
+     * */
+    handleWeekCountReceived(message) {
+        if (message.data.length !== 0) {
+            let [[weekCount]] = message.data;
+            weekCount = weekCount < 10 ? ('0' + weekCount).slice(-2) : weekCount;
+            this.setState({ weekCount });
+        }
+    }
+
+    /**
+     * @inheritDoc
+     * @returns {ReactElement} Render the APIM Api Created widget
+     * @memberof APIMApiCreated
+     */
+    render() {
+        const themeName = this.props.muiTheme.name;
+        const {
+            localeMessages, faultyProviderConf, totalCount, weekCount,
+        } = this.state;
+
+        if (localeMessages) {
+            if (faultyProviderConf) {
+                return (
+                    <IntlProvider locale={languageWithoutRegionCode} messages={localeMessages}>
+                        <div
+                            style={{
+                                margin: 'auto',
```

Comment:
Since we are following css-in-js [1], lets avoid inline styles. So move these styles to the `styles` object.

[1] https://material-ui.com/css-in-js/basics/

Reply:
Applies for other places as well.

Reply:
Have to go with this approach since we can't wrap the widget with higher order component 'withStyles'.